zallet/build.rs

```diff
@@ -137,3 +150,78 @@ impl Cli {
         Ok(())
     }
 }
+
+fn generate_rpc_help(out_dir: &Path) -> Result<(), Box<dyn Error>> {
+    // Parse the source file containing the `Rpc` trait.
+    let methods_rs = fs::read_to_string(JSON_RPC_METHODS_RS)?;
+    let methods_ast = syn::parse_file(&methods_rs)?;
+
+    let rpc_trait = methods_ast
+        .items
+        .iter()
+        .find_map(|item| match item {
+            syn::Item::Trait(item_trait) if item_trait.ident == "Rpc" => Some(item_trait),
+            _ => None,
+        })
+        .expect("present");
+
+    let mut contents = "static COMMANDS: ::phf::Map<&str, &str> = ::phf::phf_map! {\n".to_string();
+
+    for item in &rpc_trait.items {
+        match item {
+            syn::TraitItem::Fn(method) => {
+                // Find methods via their `#[method(name = "command")]` attribute.
+                let mut command = None;
+                method
+                    .attrs
+                    .iter()
+                    .find(|attr| attr.path().is_ident("method"))
+                    .and_then(|attr| {
+                        attr.parse_nested_meta(|meta| {
+                            command = Some(meta.value()?.parse::<syn::LitStr>()?.value());
+                            Ok(())
+                        })
+                        .ok()
+                    });
+
+                if let Some(command) = command {
+                    contents.push('"');
+                    contents.push_str(&command);
+                    contents.push_str("\" => \"");
+
+                    for attr in method
+                        .attrs
+                        .iter()
+                        .filter(|attr| attr.path().is_ident("doc"))
+                    {
+                        if let syn::Meta::NameValue(doc_line) = &attr.meta {
+                            if let syn::Expr::Lit(docs) = &doc_line.value {
+                                if let syn::Lit::Str(s) = &docs.lit {
+                                    // Trim the leading space from the doc comment line.
+                                    let line = s.value();
+                                    let trimmed_line =
+                                        if line.is_empty() { &line } else { &line[1..] };
+
+                                    let escaped = trimmed_line.escape_default().collect::<String>();
+
+                                    contents.push_str(&escaped);
+                                    contents.push_str("\\n");
+                                }
+                            }
+                        }
+                    }
+
+                    contents.push_str("\",\n");
+                }
+            }
```

Comment:
It looks like this could be simplified but i am not sure if it will worth the effort. If you are interested i can try to send a suggestion.